src/Tasks/Exec.cs

```diff
@@ -611,12 +611,22 @@ protected internal override void AddCommandLineCommands(CommandLineBuilderExtens
 
                     // If for some crazy reason the path has a & character and a space in it
                     // then get the short path of the temp path, which should not have spaces in it
-                    // and then escape the &
                     if (batchFileForCommandLine.Contains("&") && !batchFileForCommandLine.Contains("^&"))
                     {
                         batchFileForCommandLine = NativeMethodsShared.GetShortFilePath(batchFileForCommandLine);
-                        batchFileForCommandLine = batchFileForCommandLine.Replace("&", "^&");
                     }
+
+                    StringBuilder fileName = StringBuilderCache.Acquire(batchFileForCommandLine.Length).Append(batchFileForCommandLine);
+
+                    // Escape any '(', ')', or '&'
+                    for(int i = 1; i < fileName.Length; i++)
+                    {
+                        if((fileName[i] == '(' || fileName[i] == ')' || fileName[i] == '&') && fileName[i-1] != '^')
+                        {
+                            fileName.Insert(i++, '^');
```

Comment:
Why not append each character individually so you don't have to insert into the middle?

Also, why not start at 0?

Reply:
also cache `fileName[i]` in a local char variable to save on access
add spaces around `[i - 1]` or format this area using Format Selection or Format Document

Reply:
Not starting at 0 here because we're on windows and `batchFileForComandLine` points to a `.cnd` file in the temp folder. It really should start at 3 to skip over `c:/`.